src/platforms/dialogflow/DialogFlowEvent.ts

```diff
@@ -84,14 +85,19 @@ export class DialogFlowEvent extends VoxaEvent {
    * After that we'll default to the userStorage value
    */
   protected initUser(): void {
+    const { originalDetectIntentRequest } = this.rawEvent;
     const { conv } = this.google;
     const storage = conv.user.storage as any;
     let userId: string;
 
+    this.source = _.get(originalDetectIntentRequest, "source", "google");
+
     if (_.get(storage, "voxa.userId")) {
       userId = storage.voxa.userId;
     } else if (conv.user.id) {
       userId = conv.user.id;
+    } else if (this.source === "facebook") {
+      userId = _.get(originalDetectIntentRequest, "payload.data.sender.id", v1());
```

Comment:
Seems like this code right here is the reason you marked `userId` as possibly undefined, i think the following should fix that 

```suggestion
      userId = _.get(originalDetectIntentRequest, "payload.data.sender.id") || v1();
```